testing.md

```diff
@@ -0,0 +1,40 @@
+# Testing
+
+We currently have some tests in place, however, we are looking to improve our testing coverage. So any contributions are welcome in that regard, when writing tests, make sure to read through our testing guidelines in the section below
+
+# Guidelines
+
+## Front-end Tests
+
+For writing good Vue.js tests, you should consider the following: 
+
+- Tests should be written in a way that is easy to understand and maintain.
+- Prioritize Unit tests rather than Integration or End-to-End tests, as they are easier to maintain and debug.
+- Tests should mock the functionality of the components they are testing, and not the functionality of the components they are dependent on. Make sure to keep them as separate as possible
+
+Feel free to implement the testing strategy you see fit, some testing strategies may include:
+
+- **Mocking** the functionality of the components by replacing the dependencies and services with fake implementations, which is valuable for isolating behaviour
+  
+- **Stubbing** similar to Mocking, but by replacing a function or method with a fake implementation of it, returning a predetermined value
+  
+- **Component Snapshot Testing** which would involve testing a component's rendered output and comparing it to a previously saved snapshot. 
+
+For further reference to writing Vue tests, please refer to the Vue Testing Handbook [here](https://lmiller1990.github.io/vue-testing-handbook/).
+
+## Back-end Tests
+
+For writing good Express tests, you should consider the following:
```

Comment:
Mention that ALL files with the extension *.spec.js are run by "npm run test" BUT for github actions we can not expect there to be a database attached. So non-database-dependent tests need to be in files with *.system.spec.js" file extension.  Only these files are run by GitHub actions.